src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+                                LOG.warn("stop_code is not present in file {}/{}. Reverting to stop_id", feedIndex, feedsToMerge.size());
+                                // If the key value for stop_code is not present, revert to stop_id.
+                                keyField = table.getKeyFieldName();
+                                keyFieldIndex = table.getKeyFieldIndex(fieldsFoundInZip);
+                                keyValue = csvReader.get(keyFieldIndex);
+                                stopCodeMissingFromFirstTable = true;
+                            }
+                        } else {
+                            // Check whether stop_code exists for the subsequent files.
+                            String firstStopCodeValue = csvReader.get(getFieldIndex(fieldsFoundInZip, "stop_code"));
+                            if (stopCodeMissingFromFirstTable && !"".equals(firstStopCodeValue)) {
+                                // If stop_code was missing from the first file and exists for the second, we consider
+                                // that a failing error.
+                                mergeFeedsResult.failed = true;
+                                mergeFeedsResult.errorCount++;
+                                mergeFeedsResult.failureReasons.add(
+                                    "If one stops.txt file contains stop_codes, both feed versions must stop_codes."
+                                );
+                            }
+                        }
+                    }
+                    boolean skipRecord = false;
+                    String[] rowValues = new String[specFields.size()];
+                    String[] values = csvReader.getValues();
+                    if (values.length == 1) {
+                        LOG.warn("Found blank line. Skipping...");
+                        continue;
+                    }
+                    // Piece together the row to write, which should look practically identical to the original
+                    // row except for the identifiers receiving a prefix to avoid ID conflicts.
+                    for (int specFieldIndex = 0; specFieldIndex < specFields.size(); specFieldIndex++) {
+                        Field field = specFields.get(specFieldIndex);
+                        // Get index of field from GTFS spec as it appears in feed
+                        int index = fieldsFoundList.indexOf(field);
+                        String val = csvReader.get(index);
+                        // Determine if field is a GTFS identifier.
+                        boolean isKeyField = field.isForeignReference() || keyField.equals(field.name);
+                        Set<NewGTFSError> idErrors = table.checkReferencesAndUniqueness(keyValue, lineNumber, field, val, referenceTracker, keyField, orderField);
+                        // Store values for key fields that have been encountered.
+                        switch (table.name) {
```

Comment:
I'm not a huge fan of this switch statement that creates yet another level of nesting. Is there a way the [Strategy Pattern](https://en.wikipedia.org/wiki/Strategy_pattern) could be used here instead?

Reply:
Consider this ask very low-priority.  ;)